packages/astro/src/core/build/static-build.ts

```diff
@@ -418,27 +418,26 @@ async function ssrMoveAssets(opts: StaticBuildOptions) {
 }
 
 /**
- * This function takes as input the virtual module name of an astro page and transform
- * to generate an `.mjs` file:
+ * This function takes the virtual module name of any page entrypoint and
+ * transforms it to generate a final `.mjs` output file.
  *
  * Input: `@astro-page:src/pages/index@_@astro`
- *
  * Output: `pages/index.astro.mjs`
+ * Input: `@astro-page:../node_modules/my-dep/injected.astro@_@astro`
+ * Output: `pages/injected.mjs`
  *
- * 1. We remove the module id prefix, `@astro-page:`
- * 2. We remove `src/`
- * 3. We replace square brackets with underscore, for example `[slug]`
- * 4. At last, we replace the extension pattern with a simple dot
- * 5. We append the `.mjs` string, so the file will always be a JS file
+ * 1. We clean the `facadeModuleId` by removing the `@astro-page:` prefix and `@_@astro` suffix
+ * 2. We find the matching route pattern in the manifest (or fallback to the cleaned module id)
+ * 3. We replace square brackets with underscore (`[slug]` => `_slug_`) and `...` with `` (`[...slug]` => `_---slug_`).
+ * 4. We append the `.mjs` extension, so the file will always be an ESM module
  *
- * @param facadeModuleId
+ * @param facadeModuleId string
+ * @param pages AllPagesData
  */
-function makeAstroPageEntryPointFileName(facadeModuleId: string) {
-	return `${facadeModuleId
+function makeAstroPageEntryPointFileName(facadeModuleId: string, pages: AllPagesData) {
+	const pageModuleId = `${facadeModuleId
 		.replace(ASTRO_PAGE_RESOLVED_MODULE_ID, '')
-		.replace('src/', '')
-		.replaceAll('[', '_')
-		.replaceAll(']', '_')
-		// this must be last
-		.replace(ASTRO_PAGE_EXTENSION_POST_PATTERN, '.')}.mjs`;
+		.replace(ASTRO_PAGE_EXTENSION_POST_PATTERN, '.')}`;
+	const name = pages[pageModuleId]?.route?.route ?? pageModuleId;
```

Comment:
I think this must be done at the end? Here `pageModuleId` has its name still mangled and it won't match any route